src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -29,6 +61,23 @@ public TypeSyntaxSimplifierWalker(CSharpSimplifyTypeNamesDiagnosticAnalyzer anal
             _cancellationToken = cancellationToken;
         }
 
+        public override void VisitUsingDirective(UsingDirectiveSyntax node)
+        {
+            if (node.Alias is object)
+            {
+                if (node.Name.GetRightmostName() is IdentifierNameSyntax identifierName)
+                {
+                    var identifierAlias = identifierName.Identifier.ValueText;
+                    if (!RoslynString.IsNullOrEmpty(identifierAlias))
+                    {
+                        ImmutableInterlocked.Update(ref _aliasedNames, (set, alias) => set.Add(alias), identifierAlias);
```

Comment:
Do we expect this walker to be executed from multiple threads? Otherwise, can we not just a simple mutable hash set as the field and add to it here?

Reply:
I've actually rewritten this in a follow-up commit. I'll incorporate it.

Reply:
Done!